### PR TITLE
upgrade error handling

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

This PR:
- improves the error handler log messages
- refactors the error handling logic in aea
- refactors constants
- minor other fixes

## Fixes

-

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

-

### CODE REVIEW

File: aea/aea.py [hunk 1]
```diff
@@ -187,12 +181,15 @@ def _handle(self, envelope: Envelope) -> None:
...
-        # TODO make this working for different skill/protocol versions.
+        # TODO specify error handler in config and make this work for different skill/protocol versions.
         error_handler = self.resources.get_handler(
-            DefaultMessage.protocol_id, ERROR_SKILL_ID,
+            DefaultMessage.protocol_id, DEFAULT_SKILL
         )
 
-        assert error_handler is not None, "ErrorHandler not initialized"
+        if error_handler is None:
+            logger.warning("ErrorHandler not initialized. Stopping AEA!")
```
Comment (DavidMinarsch) [inline aea/aea.py:190]: Throwing an assertion error here is messy, better to stop the AEA gracefully

File: aea/configurations/constants.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+# ------------------------------------------------------------------------------
+
+"""Module to declare constants."""
+
+from pathlib import Path
+
+from aea.configurations.base import PublicId
+from aea.crypto.fetchai import FETCHAI
+
+DEFAULT_CONNECTION = PublicId.from_str("fetchai/stub:0.1.0")
```
Comment (DavidMinarsch) [inline aea/configurations/constants.py:27]: All public ids should be specified in string form, makes it easier to do string substitution

File: aea/skills/error/handlers.py [hunk 3]
```diff
@@ -87,45 +93,22 @@ def send_decoding_error(self, envelope: Envelope) -> None:
         """
         Handle a decoding error.
 
-        :param envelope: the envelope
-        :return: None
-        """
```
Comment (DavidMinarsch) [inline aea/skills/error/handlers.py:92]: unused

File: aea/aea.py (hunk 1 above)
Comment (marcofavorito) [reply aea/aea.py:190]: ok

Comment (marcofavorito) [reply aea/aea.py:190]: maybe do a `try except` on the whole `_handle` method?

File: aea/configurations/constants.py (hunk 2 above)
Comment (marcofavorito) [reply aea/configurations/constants.py:27]: yes.

File: aea/aea.py (hunk 1 above)
Comment (DavidMinarsch) [reply aea/aea.py:190]: Why would we do that? Seems a bit brute force.

**Pull request merged**